pypowsybl/shortcircuit.py

```diff
@@ -0,0 +1,251 @@
+        c_parameters.with_fortescue_result = False
+        c_parameters.min_voltage_drop_proportional_threshold = 0
+        c_parameters.provider_parameters_keys = []
+        c_parameters.provider_parameters_values = []
+        return c_parameters
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}(" \
+               f"with_feeder_result={self.with_feeder_result!r}" \
+               f", with_limit_violations={self.with_limit_violations!r}" \
+               f", study_type={self.study_type!r}" \
+               f")"
+
+class ShortCircuitAnalysisResult:
+    """
+    The result of a short-circuit analysis.
+    """
+
+    def __init__(self, handle: _pypowsybl.JavaHandle):
+        self._handle = handle
+
+    @property
+    def fault_results(self) -> _pd.DataFrame:
+        """
+        contains the results, for each fault, in a dataframe representation.
+        """
+        return _create_data_frame_from_series_array(_pypowsybl.get_fault_results(self._handle))
+
+    @property
+    def feeder_results(self) -> _pd.DataFrame:
+        """
+        contains the contributions of each feeder to the short circuit current, in a dataframe representation.
+        """
+        return _create_data_frame_from_series_array(_pypowsybl.get_feeder_results(self._handle))
+
+    @property
+    def limit_violations(self) -> _pd.DataFrame:
+        """
+        contains a list of all the violations after the fault, in a dataframe representation.
+        """
+        return _create_data_frame_from_series_array(_pypowsybl.get_short_circuit_limit_violations(self._handle))
+
+
+class ShortCircuitAnalysis():
+    """
+    Allows to run a short-circuit analysis on a network.
+    """
+
+    def __init__(self, handle: _pypowsybl.JavaHandle):
+        self._handle = handle
+
+
+    def _set_faults(self, fault_type: ShortCircuitFaultType, dfs: _List[_Optional[_DataFrame]],
+                    **kwargs: _ArrayLike) -> None:
+        metadata = _pypowsybl.get_faults_dataframes_metadata(fault_type)
+        c_dfs = _get_c_dataframes(dfs, [metadata], **kwargs)
+        _pypowsybl.set_faults(self._handle, c_dfs[0], fault_type)
+
+
+    def set_faults(self, df: _DataFrame = None, **kwargs: _ArrayLike) -> None:
+        """
+        Define faults to be analysed in the short-circuit simulation.
+
+        Args:
+            df: Attributes as a dataframe.
+            kwargs: Attributes as keyword arguments.
+
+        Notes:
+            The current implementation allows the simulation of three-phased bus faults, where
+            the fault resistance and reactance are connected to the ground in series.
+
+            Data may be provided as a dataframe or as keyword arguments.
+            In the latter case, all arguments must have the same length.
+
+            Valid attributes are:
+
+            - **id**: the id of the fault.
+            - **element_id**: the id of the bus on which the fault will be simulated (bus/view topology).
```

Comment:
I was wondering should it be bus_id instead of element_id?

Reply:
good point. Thx. I'll change it to bus_id.
In the pypowsybl's context. bus_id is more meaningful
(it is "element_id" in the shortcircuit APIs,  and I kept it that way).

Reply:
@colinepiloquet is it ok ?

Reply:
I agree with you, for now it seems to make more sense to put bus_id. But I think it would be easier to have very similar input dataframes for bus and branch faults

Reply:
let's keep the more generic 'element_id', for the time being